google/earthenterprise
# Python 3 Migration: Cutter Bug fix

Fixes import and unicode issues stemming from the python 3 migration. Do not merge until the Python 3 Server migration PR is merged.

To test:

1. Build and install according to the build docs (https://github.com/tst-cjeffries/earthenterprise/blob/python_3_cutter_bug/earth_enterprise/BUILD_RHEL_CentOS.md)
2. Run the tutorials.
    `sudo /opt/google/share/tutorials/fusion/download_tutorial.sh`
    `sudo cp earthenterprise/earth_enterprise/src/fusion/tools/gee_test/tutorial_files/providers.xml /gevol/assets/.userdata/`
    `sudo env "PATH=$PATH:/opt/google/bin" earthenterprise/earth_enterprise/src/fusion/tools/gee_test/run_tutorial.sh`
    Then wait until the tutorials are finished building. This can be monitored in with the status scripts in that same directory.
    `sudo env "PATH=$PATH:/opt/google/bin" earthenterprise/earth_enterprise/src/fusion/tools/gee_test/publish_tutorial.sh`
3. `sudo /opt/google/bin/gecutter enable`
4. Launch cutter from the admin page and make both a 2d and 3d cut.
5. Verify the 2d and 3d cut by publishing them on the admin page and viewing them.
6. Run the glc assembly and disassembly on the 3d cut.
7. Verify the assembled glc by registering and viewing it.
8. Verify the disassembled glb by either moving it into `/opt/google/gehttpd/htdocs/cutter/globes`, registering it and viewing it or by loading it into portable.
